### PR TITLE
Add VS Code bits and fix errors

### LINKED ISSUE #489: [WebToolsE2E][Aspire][Document] Tutorial: Use .NET Aspire messaging components in ASP.NET Core - need some updates
**Issue description**

Using the latest Aspire bits (VS 17.10 P2 build), we tried running through the [Tutorial: Use .NET Aspire messaging components in ASP.NET Core](https://learn.microsoft.com/en-us/dotnet/aspire/messaging/messaging-components?tabs=connection-string) document. Chose 'Connection String' option.

Some updates are needed.

**Prerequisites**

- [ ] We should add a step of Azure CLI install for this section. Because the last step 'Clean up resources' need to run the Azure CLI command.

**Add the Worker Service project**

- [ ]  For the sample code here, it should be `aspiremessaging-workerservice`, not the `aspiremessaging.workerservice`:

_Document_:
image
_Actual_:
image

**Create the API endpoint**

- [ ]  The sample codes miss a semicolon in that last , it will show an error.
_Document_:
image
_Actual_:
image

**Clean up resources**

- [ ]  When we followed the section 'Set up the Azure Service Bus account', it gives us 2 different options to set up the account, one is Azure portal, another one is Azure CLI, but for this section, there is only one option 'Run the following Azure CLI command...' to delete the resource group, should we also indicate that we can delete the resource group from Azure portal?

---
[Associated WorkItem - 261689](https://dev.azure.com/msft-skilling/Content/_workitems/edit/261689)

### PR DESCRIPTION
**Pull request by IEvangelist**
## Summary

Add VS Code bits and fix errors.

Fixes #489

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/messaging/messaging-components.md](https://github.com/dotnet/docs-aspire/blob/96885b2f1e76b04cbc2e47c4a906a926874d0e9a/docs/messaging/messaging-components.md) | [Tutorial: Use .NET Aspire messaging components in ASP.NET Core](https://review.learn.microsoft.com/en-us/dotnet/aspire/messaging/messaging-components?branch=pr-en-us-1043) |

### CODE REVIEW

File: docs/messaging/messaging-components.md [hunk 1]
```diff
@@ -166,22 +268,24 @@ This method accomplishes the following tasks:
 - Registers a <xref:Microsoft.Azure.Commands.ServiceBus.ServiceBusClient> with the DI container for connecting to Azure Service Bus.
 - Automatically enables corresponding health checks, logging, and telemetry for the respective services.
 
-In the _:::no-loc text="appsettings.json":::_ file of the `AspireMessaging` project, add the corresponding connection information:
+In the _:::no-loc text="appsettings.json":::_ file of the same project, add the corresponding connection information:
 
 # [Passwordless (Recommended)](#tab/passwordless)
 
-```json
+```jsonc
```
Comment (IEvangelist) [inline docs/messaging/messaging-components.md:275]:
```suggestion
```json
```

File: docs/messaging/messaging-components.md [hunk 2]
```diff
@@ -166,22 +268,24 @@ This method accomplishes the following tasks:
...
-    "serviceBusConnectionName": "{your_namespace}.servicebus.windows.net"
+    "serviceBusConnection": "{your_namespace}.servicebus.windows.net"
   }
 }
 ```
 
 # [Connection String](#tab/connection-string)
 
-```json
+```jsonc
```
Comment (IEvangelist) [inline docs/messaging/messaging-components.md:286]:
```suggestion
```json
```

File: docs/messaging/messaging-components.md [hunk 3]
```diff
@@ -221,43 +341,54 @@ app.MapPost("/notify", static async (ServiceBusClient client, string message) =>
...
 - Registers a <xref:Microsoft.Azure.Commands.ServiceBus.ServiceBusClient> with the DI container for connecting to Azure Service Bus.
 - Automatically enables corresponding health checks, logging, and telemetry for the respective services.
 
-In the _:::no-loc text="appsettings.json":::_ file of the `AspireMessaging.WorkerService` project, add the corresponding connection information:
+In the _:::no-loc text="appsettings.json":::_ file of the **AspireSample.WorkerService** project, add the corresponding connection information:
 
 # [Passwordless (Recommended)](#tab/passwordless)
 
-```json
+```jsonc
```
Comment (IEvangelist) [inline docs/messaging/messaging-components.md:378]:
```suggestion
```json
```

File: docs/messaging/messaging-components.md [hunk 4]
```diff
@@ -221,43 +341,54 @@ app.MapPost("/notify", static async (ServiceBusClient client, string message) =>
...
-    "serviceBusConnectionName": "{your_namespace}.servicebus.windows.net"
+    "serviceBusConnection": "{your_namespace}.servicebus.windows.net"
   }
 }
 ```
 
 # [Connection String](#tab/connection-string)
 
-```json
+```jsonc
```
Comment (IEvangelist) [inline docs/messaging/messaging-components.md:389]:
```suggestion
```json
```

File: docs/messaging/messaging-components.md [hunk 5]
```diff
@@ -17,6 +18,8 @@ Cloud-native apps often require scalable messaging solutions that provide capabi
 
 [!INCLUDE [aspire-prereqs](../includes/aspire-prereqs.md)]
 
+In addition to the preceding prerequisites, you'll also need to install the Azure CLI. To install the Azure CLI, follow the instructions in the [Azure CLI installation guide](/cli/azure/install-azure-cli).
```
Comment (gewarren) [inline docs/messaging/messaging-components.md:21]:
```suggestion
In addition to the preceding prerequisites, you also need to install the Azure CLI. To install the Azure CLI, follow the instructions in the [Azure CLI installation guide](/cli/azure/install-azure-cli).
```

File: docs/messaging/messaging-components.md [hunk 6]
```diff
@@ -60,102 +63,201 @@ Retrieve the connection string for your Service Bus namespace from the **Shared
...
+## Add the .NET Aspire component to the API
 
-:::image type="content" source="../media/messaging-project.png" alt-text="A screenshot showing the completed sample .NET Aspire app structure.":::
+Add the [.NET Aspire Azure Service Bus](azure-service-bus-component.md) component to your **AspireSample.ApiService** app:
 
-## Add the .NET Aspire component to the API
+:::zone pivot="visual-studio"
 
-Add the [.NET Aspire Azure Service Bus](azure-service-bus-component.md) component to your `AspireMessaging.ApiService` app:
+1. In the **Solution Explorer**, double-click the **AspireSample.ApiService.csproj** file to open it's XML file.
```
Comment (gewarren) [inline docs/messaging/messaging-components.md:227]:
```suggestion
1. In the **Solution Explorer**, double-click the **AspireSample.ApiService.csproj** file to open its XML file.
```

File: docs/messaging/messaging-components.md [hunk 7]
```diff
@@ -60,102 +63,201 @@ Retrieve the connection string for your Service Bus namespace from the **Shared
...
+1. Add the following `<PackageReference>` element to the `<ItemGroup>` element:
```
Comment (gewarren) [inline docs/messaging/messaging-components.md:228]:
```suggestion
1. Add the following `<PackageReference>` item to the `<ItemGroup>` element:
```

File: docs/messaging/messaging-components.md [hunk 8]
```diff
@@ -60,102 +63,201 @@ Retrieve the connection string for your Service Bus namespace from the **Shared
...
+    ```
+
+:::zone-end
+:::zone pivot="vscode"
+
+1. In the **Solution Explorer**, right-click on the **AspireSample.ApiService** project and select **Add NuGet Package**:
+
+    :::image type="content" source="media/vscode-add-nuget-package.png" lightbox="media/vscode-add-nuget-package.png" alt-text="Visual Studio Code: Add NuGet package to AspireSample.ApiService project.":::
+
+1. Enter **Aspire.Azure.Messaging.ServiceBus** in the search box and select the package from the list:
```
Comment (gewarren) [inline docs/messaging/messaging-components.md:244]:
```suggestion
1. Enter **Aspire.Azure.Messaging.ServiceBus** in the search box and select the package from the list.
```

File: docs/messaging/messaging-components.md [hunk 9]
```diff
@@ -221,35 +341,45 @@ app.MapPost("/notify", static async (ServiceBusClient client, string message) =>
...
+
+app.MapDefaultEndpoints();
+
+app.Run();
 ```
 
 ## Add the .NET Aspire component to the Worker Service
 
-Add the [.NET Aspire Azure Service Bus](azure-service-bus-component.md) component to your `AspireMessaging.WorkerService` app:
+Add the [.NET Aspire Azure Service Bus](azure-service-bus-component.md) component to your **AspireSample.WorkerService** project. Follow the same steps as you did before when you added the [Aspire.Azure.Messaging.ServiceBus](https://nuget.org/packages/Aspire.Azure.Messaging.ServiceBus) NuGet package to the **AspireSample.ApiService** project. Once it's been added, you can configure the worker service to process messages from the Service Bus topic.
```
Comment (gewarren) [inline docs/messaging/messaging-components.md:352]: What's the Service Bus "topic"?

File: docs/messaging/messaging-components.md [hunk 10]
```diff
@@ -330,17 +474,37 @@ public class Worker(
 
 The sample app is now ready for testing. Verify that the data submitted to the API is sent to the Azure Service Bus topic and consumed by the subscriber worker service:
 
-1. Launch the Aspire app by selecting the run button (Visual Studio) or running `dotnet run --project AspireMessaging.AppHost`. The .NET Aspire dashboard app should open in the browser.
-1. On the resources page, in the **apiservice** row, find the link in the **Endpoints** that opens the `weatherforecast` endpoint. Note the HTTPS port number.
-1. On the .NET Aspire dashboard, navigate to the logs for the **aspiremessaging-workerservice** project.
-1. In a terminal window, use the `curl` command to send a test message to the API:
+:::zone pivot="visual-studio"
+
+1. Launch the .NET Aspire app by selecting the run button **Start** debugging button, or by pressing <kbd>F5</kbd>. The .NET Aspire dashboard app should open in the browser.
```
Comment (gewarren) [inline docs/messaging/messaging-components.md:479]:
```suggestion
1. Launch the .NET Aspire app by selecting the **Start** debugging button, or by pressing <kbd>F5</kbd>. The .NET Aspire dashboard app should open in the browser.
```

File: docs/messaging/messaging-components.md [hunk 11]
```diff
@@ -330,17 +474,37 @@ public class Worker(
...
+
+:::zone-end
+:::zone pivot="vscode"
+
+1. Launch the .NET Aspire app by selecting the run button **Start** debugging button, or by pressing <kbd>F5</kbd>. The .NET Aspire dashboard app should open in the browser.
```
Comment (gewarren) [inline docs/messaging/messaging-components.md:484]:
```suggestion
1. Launch the .NET Aspire app by selecting the **Start** debugging button, or by pressing <kbd>F5</kbd>. The .NET Aspire dashboard app should open in the browser.
```

**Review by gewarren — APPROVED** [review]

File: docs/messaging/messaging-components.md (hunk 9 above)
Comment (IEvangelist) [reply docs/messaging/messaging-components.md:352]: Service Bus has the concept of "topics", in addition to "queues" and "subs": https://learn.microsoft.com/azure/service-bus-messaging/service-bus-queues-topics-subscriptions. Folks who are familiar with Service Bus will know what that is. Do you think it needs more info?

**Pull request merged**